crates/weaver_resolved_schema/src/v2/mod.rs

```diff
@@ -0,0 +1,604 @@
+    fix_group_id("span.", group_id)
+}
+
+/// Converts a V1 registry + catalog to V2.
+pub fn convert_v1_to_v2(
+    c: crate::catalog::Catalog,
+    r: crate::registry::Registry,
+) -> Result<(Catalog, Registry), crate::error::Error> {
+    // When pulling attributes, as we collapse things, we need to filter
+    // to just unique.
+    let attributes: HashSet<attribute::Attribute> = c
+        .attributes
+        .iter()
+        .cloned()
+        .map(|a| {
+            attribute::Attribute {
+                key: a.name,
+                r#type: a.r#type,
+                examples: a.examples,
+                common: CommonFields {
+                    brief: a.brief,
+                    note: a.note,
+                    // TODO - Check this assumption.
+                    stability: a
+                        .stability
+                        .unwrap_or(weaver_semconv::stability::Stability::Alpha),
+                    deprecated: a.deprecated,
+                    annotations: a.annotations.unwrap_or_default(),
+                },
+            }
+        })
+        .collect();
+
+    let v2_catalog = Catalog::from_attributes(attributes.into_iter().collect());
+
+    // Create a lookup so we can check inheritance.
+    let mut group_type_lookup = HashMap::new();
+    for g in r.groups.iter() {
+        let _ = group_type_lookup.insert(g.id.clone(), g.r#type.clone());
+    }
+    // Pull signals from the registry and create a new span-focused registry.
+    let mut spans = Vec::new();
+    let mut span_refinements = Vec::new();
+    let mut metrics = Vec::new();
+    let mut metric_refinements = Vec::new();
+    let mut events = Vec::new();
+    let mut event_refinements = Vec::new();
+    let mut entities = Vec::new();
+    for g in r.groups.iter() {
+        let extend_type = g.extends.as_ref().and_then(|id| group_type_lookup.get(id));
+        match g.r#type {
+            GroupType::Span => {
+                // Check if we extend another span.
+                let is_refinement = g
+                    .lineage
+                    .as_ref()
+                    .and_then(|l| l.extends_group.as_ref())
+                    .and_then(|parent| group_type_lookup.get(parent))
+                    .map(|t| *t == GroupType::Span)
+                    .unwrap_or(false);
+                // Pull all the attribute references.
+                let mut span_attributes = Vec::new();
+                for attr in g.attributes.iter().filter_map(|a| c.attribute(a)) {
+                    if let Some(a) = v2_catalog.convert_ref(attr) {
+                        span_attributes.push(span::SpanAttributeRef {
+                            base: a,
+                            requirement_level: attr.requirement_level.clone(),
+                            sampling_relevant: attr.sampling_relevant.clone(),
+                        });
+                    } else {
+                        // TODO logic error!
+                    }
```

Comment:
Is this dance, attribute-ref -> attribute -> attribute-ref, necessary because the comparison rules for attributes differ between v1 and v2?
That brings us back to my previous question about the rationale behind this difference. In any case, once we've fully migrated to v2, we shouldn't have this kind of thing anymore.